v2/tools/generator/internal/astmodel/resource_type.go

```diff
@@ -621,23 +623,47 @@ func (resource *ResourceType) AsDeclarations(codeGenerationContext *CodeGenerati
 	}
 
 	var declarations []dst.Decl
+	var errs []error
 	declarations = append(declarations, resourceDeclaration)
 	declarations = append(declarations, resource.InterfaceImplementer.AsDeclarations(codeGenerationContext, declContext.Name, nil)...)
-	declarations = append(declarations, resource.generateMethodDecls(codeGenerationContext, declContext.Name)...)
+
+	decls, err := resource.generateMethodDecls(codeGenerationContext, declContext.Name)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
```

Comment:
Can also remove this else branch in case of checking error above on line 632

Reply:
Done.